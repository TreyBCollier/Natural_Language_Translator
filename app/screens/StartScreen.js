import React from "react";
import {
    Text,
    View,
    Image,
    ImageBackground,
    SafeAreaView,
    TouchableOpacity,
    StatusBar
} from "react-native";
import styles from "../styles/homeScreeen"

class StartScreen extends React.Component {
  constructor(props) {
    super(props);
    // Setting state variables
    this.state = {
    };
  }

  render() {

    return (
        <View style={styles.container}>
        {/* Setting background image for home screen */}
        <ImageBackground
          style={styles.backgroundImage}
          resizeMode="cover"
          source={require("../assets/paris.jpeg")}
        >
          <SafeAreaView style={[styles.imgContainer]}>
            {/* Adding logo to homescreen */}
            <Image
              style={styles.logo}
              source={require("../assets/logo_white.png")}
            ></Image>

            {/* Button to navigate to 'English to French' translating */}
            <View style={{width: "100%", height: "100%", position: "absolute", top: "80%"}}>
            <TouchableOpacity
              activeOpacity={0.8}
              onPress={() => this.props.navigation.navigate("EngToFre")}
              style={[styles.buttonContainer, styles.signupBtn]}
            >
              <Text style={styles.appButtonText}>Translate English to French</Text>
            </TouchableOpacity>

            {/* Button to navigate to 'French to English' translating */}
            <TouchableOpacity
              activeOpacity={0.8}
              onPress={() => this.props.navigation.navigate("FreToEng")}
              style={[styles.buttonContainer, styles.signupBtn]}
            >
              <Text style={styles.appButtonText}>Translate French to English</Text>
            </TouchableOpacity>
            </View>

          </SafeAreaView>
        </ImageBackground>
        <StatusBar style="auto" />
      </View>
    );
  }
}

export default StartScreen;