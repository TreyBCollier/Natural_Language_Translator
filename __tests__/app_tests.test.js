import 'react-native';
import React from 'react';
import StartScreen from "../app/screens/StartScreen";

import renderer from 'react-test-renderer';

  it('StartScreen renders correctly', () => {
    const tree0 = renderer.create(
      <StartScreen />
      ).toJSON();
    expect(tree0).toMatchSnapshot();
  });

 