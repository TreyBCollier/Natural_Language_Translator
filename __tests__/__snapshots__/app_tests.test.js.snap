// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignContent": "center",
        "bottom": 0,
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "testUri": "../../../app/assets/paris.jpeg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/logo_white.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": "30%",
            "marginBottom": "-2.5%",
            "resizeMode": "contain",
            "width": "90%",
          }
        }
      />
      <View
        style={
          Object {
            "height": "100%",
            "position": "absolute",
            "top": "80%",
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1.5,
              "elevation": 8,
              "height": "5%",
              "justifyContent": "center",
              "marginTop": 15,
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 0,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#fff",
                "fontSize": 27,
                "fontWeight": "bold",
              }
            }
          >
            Translate English to French
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1.5,
              "elevation": 8,
              "height": "5%",
              "justifyContent": "center",
              "marginTop": 15,
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 0,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#fff",
                "fontSize": 27,
                "fontWeight": "bold",
              }
            }
          >
            Translate French to English
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
